### PR TITLE
{Compute} Remove credentials from recordings

### PR DESCRIPTION
**Pull request by jiasli**
#14344, #14532 contains access tokens (credentials) in `x-ms-authorization-auxiliary` headers.

`x-ms-authorization-auxiliary` will not be recorded anymore since https://github.com/Azure/azure-python-devtools/commit/8915c3fc276242b6818462b50c9340fd925e0842, added by https://github.com/Azure/azure-python-devtools/pull/46. This change is released in `azure-devtools` 1.1.0.

Azure CLI didn't take this change until #14477 which bumped `azure-devtools` to 1.2.0 and was merged _after_ #14344, #14532, so this issue is already fixed. Re-running the test won't record `x-ms-authorization-auxiliary` anymore.

Actually, if a test is marked as `@live_only`, no recording should be generated at all! This could be improved in the test framework.

### CODE REVIEW

**Comment by Juliehzl** [comment]
@qwordy You could use `LiveSenarioTest` instead of `ScenarioTest` to avoid recording if you only want to run the test in live mode.https://github.com/Azure/azure-cli/blob/da4239a3b5ae9e2cc74785522bbb11f27b26eaf4/src/azure-cli/azure/cli/command_modules/vm/tests/latest/test_vm_commands.py#L4920

File: src/azure-cli/azure/cli/command_modules/vm/tests/latest/test_vm_commands.py [hunk 1]
```diff
@@ -4983,6 +4910,80 @@ def test_vm_cross_tenant_update(self, resource_group):
         self.cmd('group delete -n {another_rg} -y --subscription {aux_sub}')
 
 
+class VMSSCrossTenantUpdateScenarioTest(LiveScenarioTest):
+
+    @ResourceGroupPreparer(name_prefix='cli_test_vmss_update_', location='westus2')
+    def test_vmss_cross_tenant_update(self, resource_group):
```
Comment (jiasli) [inline src/azure-cli/azure/cli/command_modules/vm/tests/latest/test_vm_commands.py:4916]: Rename the test name to keep in sync with `test_vm_cross_tenant_update`:

```diff
- test_vmss_update_cross_tenant
+ test_vmss_cross_tenant_update
```

**Comment by qwordy** [comment]
> VMCrossTenantUpdateScenarioTest

It was written by XXX on 7/29/2020.

**Comment by qwordy** [comment]
I remember there is bug in `LiveSenarioTest`

**Comment by qwordy** [comment]
> 
> 
> @qwordy You could use `LiveSenarioTest` instead of `ScenarioTest` to avoid recording if you only want to run the test in live mode.
> 
> https://github.com/Azure/azure-cli/blob/da4239a3b5ae9e2cc74785522bbb11f27b26eaf4/src/azure-cli/azure/cli/command_modules/vm/tests/latest/test_vm_commands.py#L4920

Thank you so much!

**Review by qwordy — APPROVED** [review]

**Review by zhoxing-ms — APPROVED** [review]

**Pull request merged**